### PR TITLE
ls/merkle proof

### LINKED ISSUE #1328: Check storage proof approach for settlement call

### PR DESCRIPTION
**Pull request by MrPai**
close #1328

### CODE REVIEW

File: scripts/helper/src/commands/storage-proof.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        options: { relayWs, paraWs, blockAt }
+      } = actionParameters
+      const relayApi = await getRelayApi(relayWs.toString())
+      const api = await getApi(paraWs.toString())
+      const validationData = (await api.query.parachainSystem.validationData.at(
+        blockAt.toString()
+      )) as unknown as PersistedValidationData
+      console.log(JSON.stringify(validationData, null, 4))
+      const relayBlockHash = await relayApi.rpc.chain.getBlockHash(validationData.relayParentNumber)
+      const accountBytes = decodeAddress('CmNv7yFV13CMM6r9dJYgdi4UTJK7tzFEF17gmK9c3mTc2PG')
```
Comment (GopherJ) [inline scripts/helper/src/commands/storage-proof.ts:30]: may need to add another option for controller account

File: pallets/liquid-staking/Cargo.toml [hunk 2]
```diff
@@ -50,6 +52,7 @@ parachain-info                  = { git = 'https://github.com/paritytech/cumulus
 
 orml-xcm-support = { git = 'https://github.com/open-web3-stack/open-runtime-module-library.git' }
 orml-xtokens     = { git = 'https://github.com/open-web3-stack/open-runtime-module-library.git' }
+hex = "0.4.3"
```
Comment (GopherJ) [inline pallets/liquid-staking/Cargo.toml:55]: single quote is preferred

File: pallets/liquid-staking/src/lib.rs [hunk 3]
```diff
@@ -1327,6 +1345,40 @@ pub mod pallet {
...
+                &db,
+                &relay_root,
+                &key,
+            ) {
+                return result == value;
+            }
+            false
+        }
+
+        fn get_underlying_key(derivative_index: DerivativeIndex) -> Vec<u8> {
```
Comment (GopherJ) [inline pallets/liquid-staking/src/lib.rs:1370]: get_staking_ledger_key instead of adding new key like `get_current_era_key`

**Comment by MrPai** [comment]
image

**Comment by MrPai** [comment]
when run comman 'yarn start storage-proof ', It works now:
image

but some how, the root is not matched:
image
we need to find a better way offchain, choose the right root hash. @GopherJ

**Comment by MrPai** [comment]
Benchmark still has some problems, we may consider fixing it in another PR.

**Review by GopherJ — APPROVED** [review]

**Pull request merged**